stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/plugin/listener/internal/ThreadQueryListenerState.kt

```diff
@@ -59,35 +52,27 @@ internal class ThreadQueryListenerFull(
     }
 
     override suspend fun onGetRepliesRequest(messageId: String, limit: Int) {
-        val threadLogic = logic?.thread(messageId)
+        val threadLogic = logic.thread(messageId)
 
-        threadLogic?.setLoading(true)
-        val messages = messageRepository?.selectMessagesForThread(messageId, limit)
-        val parentMessage = threadLogic?.getMessage(messageId) ?: messages?.firstOrNull { it.id == messageId }
+        threadLogic.setLoading(true)
+        val messages = messageRepository.selectMessagesForThread(messageId, limit)
+        val parentMessage = threadLogic.getMessage(messageId) ?: messages.firstOrNull { it.id == messageId }
 
-        if (parentMessage != null && messages?.isNotEmpty() == true) {
-            threadLogic?.upsertMessages(messages)
-        } else {
-            val result = getRemoteMessage(messageId)
```

Comment:
No need to get remove message from the listener. Listeners should not never make API calls as they would open the possibility for infinite loops when one listener calls the other, which makes everything much more complicated. 

Also, it is not necessary to update this message as it will be updated one the original request is completed successfully.